ostree/ostree_transport.go

```diff
@@ -84,24 +84,16 @@ func NewReference(image string, repo string) (types.ImageReference, error) {
 	// image is not _really_ in a containers/image/docker/reference format;
 	// as far as the libOSTree ociimage/* namespace is concerned, it is more or
 	// less an arbitrary string with an implied tag.
-	// We use the reference.* parsers basically for the default tag name in
-	// reference.TagNameOnly, and incidentally for some character set and length
-	// restrictions.
-	var ostreeImage reference.Named
-	s := strings.SplitN(image, ":", 2)
-
-	named, err := reference.WithName(s[0])
+	// Parse the image using rereference.ParseNormalizedNamed so that we can
+	// check whether the images has a tag specified and we can add ":latest" if needed
+	// A tag must be always specified in the OSTree branch name.
+	ostreeImage, err := reference.ParseNormalizedNamed(image)
```

Comment:
“A tag must be always specified” seems incorrect, with `"busybox@sha256:…"` now supported.

`s/rereference/reference/`.

Reply:
pushed a new version with this fixed